woocommerce/woocommerce
# Switch to individual test files

### Changes proposed in this Pull Request:

This PR renames the files in `tests/e2e/specs/*/` to run each `describe` test group as a separate file. Over the last 3 months we have almost tripled the number of tests. 

In the current setup, most tests run in `shopper.test.js` or `merchant.test.js`. Our test setup runs resets before the beginning of each test file. By switching to individual tests we will have the reset run between each `describe` test group. Making this change should improve the reliability of the tests.

Initially, this PR will have a significant number of tests fail in CI. This will help us identify flaky tests which we can address in individual PRs (ensuring that the changes pass with both `trunk` and this branch). Merging trunk with this PR as those tests are fixed should lead to all tests passing in this branch.

### How to test the changes in this Pull Request:

1. Verify CI run
2. Switch to this branch
3. Run `composer install --no-dev`
4. Run `npm i`
5. Run `npm run build:assets`
6. Run `npx wc-e2e docker:up`
7. Run `npm wc-e2e test:e2e`
8. Tests should pass locally

### Changelog entry

> N/A
